cpp/density/SphereVoxelization.cc

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2010-2020 The Regents of the University of Michigan
+// This file is from the freud project, released under the BSD 3-Clause License.
+
+#include <cmath>
+#include <stdexcept>
+
+#include "SphereVoxelization.h"
+
+/*! \file SphereVoxelization.cc
+    \brief Routines for computing voxelized densities from spheres centered at points.
+*/
+
+namespace freud { namespace density {
+
+SphereVoxelization::SphereVoxelization(vec3<unsigned int> width, float r_max)
+    : m_width(width), m_r_max(r_max)
+{
+    // dummy initial box, dimensionality of the calculation will be defined by
+    // the first neighbor query passed to the compute method
+    m_box = NULL;
```

Comment:
I support the idea of preventing dimensionality changes in either direction, but I am not sure `NULL` is the right strategy here. Honestly I have no idea how `m_box` would act after initializing it to `NULL`.

`NULL` is not recommended for use in C++11 or newer. https://stackoverflow.com/questions/55520687/is-null-a-pointer

[`nullptr`](https://en.cppreference.com/w/cpp/language/nullptr) is the preferred way to use null pointers, but `m_box` _isn't a pointer_. 

I would rather keep a flag like `bool m_has_computed` that is used when checking the "new" box dimensionality against the stored box dimensionality.

Reply:
I agree with switching this to `nullptr`, but the class is designed such that nothing can happen before `compute()` is called, so I don't think we have to worry about how `m_box` acts after setting it to `nullptr` because the first thing that happens in `compute()` is that we set `m_box` to something that isn't `nullptr`.

Reply:
To clarify, I wasn’t recommending to use nullptr for this case. You should not assign a pointer type to m_box because it isn’t a pointer. I would like to be explicit with the m_has_computed flag. The m_box should be default-constructed and the flag should track its assignment. That’s a much safer approach.